pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -241,9 +241,13 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 	return c
 }
 
-func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(proxy *model.Proxy, instance *model.ServiceInstance, bind string) *cluster.Cluster {
+// buildInboundClusterForPortOrUDS constructs a single inbound listener. The cluster will be bound to `inbound|clusterPort||`, and send traffic to <bind>:<instance.Endpoint.EndpointPort>
+// A workload will have a single inbound cluster per port. In general this works properly, with the exception of the Service-oriented DestinationRule, and upstream
+// protocol selection. Our documentation currently requires a single protocol per port, and the DestinationRule issue is slated to move to Sidecar.
+// Note: clusterPort and instance.Endpoint.EndpointPort are identical for standard Services; however, Sidecar.Ingress allows these to be different.
+func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(proxy *model.Proxy, clusterPort int, instance *model.ServiceInstance, bind string) *cluster.Cluster {
 	clusterName := util.BuildInboundSubsetKey(proxy, instance.ServicePort.Name,
```

Comment:
One concern I have is this will change the behavior of https://istio.io/latest/docs/reference/config/networking/envoy-filter/#EnvoyFilter-ClusterMatch. But the `subset` and `service` were already broken by the previous change. Not sure what the best behavior here is

Reply:
It can also breaks router match

Reply:
what is router match?

Reply:
https://istio.io/latest/docs/reference/config/networking/envoy-filter/#EnvoyFilter-RouteConfigurationMatch

Reply:
As the route name is also changed to be `inbound|<workload port>||`

Reply:
This is only for the inbound route, does the route match even apply ?

Reply:
btw we already changed this from 1.7 to 1.8

Reply:
> This is only for the inbound route, does the route match even apply ?

I think we donot have limitations on inbound

>btw we already changed this from 1.7 to 1.8

With 1.8.0, isn't it the service port rather than the pod targetport?

Reply:
BTW, why do we change the format initially i missed that context

Reply:
> BTW, why do we change the format initially i missed that context


https://github.com/istio/istio/issues/26861, https://docs.google.com/document/d/1bIuwv0E98kyn8VZ_CDVGcZziIHyr3-ACiTmPqGgp5uY/

The inbound configuration is supposed to be entirely workload based; having Service oriented config (like `hostname`, previously in the cluster name) is not correct